git/attribs.go

```diff
@@ -0,0 +1,87 @@
+package git
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/git-lfs/git-lfs/tools"
+	"github.com/rubyist/tracerx"
+)
+
+const (
+	LockableAttrib = "lockable"
+)
+
+// AttributePath is a path entry in a gitattributes file which has the LFS filter
+type AttributePath struct {
+	// Path entry in the attribute file
+	Path string
+	// The attribute file which was the source of this entry
+	Source string
+	// Path also has the 'lockable' attribute
+	Lockable bool
+}
+
+// GetAttributePaths returns a list of entries in .gitattributes which are
+// configured with the filter=lfs attribute
+// workingDIr is the root of the working copy
+// gitDir is the root of the git repo
+func GetAttributePaths(workingDir, gitDir string) []AttributePath {
+	paths := make([]AttributePath, 0)
+
+	for _, path := range findAttributeFiles(workingDir, gitDir) {
+		attributes, err := os.Open(path)
+		if err != nil {
+			continue
+		}
+
+		scanner := bufio.NewScanner(attributes)
+
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.Contains(line, "filter=lfs") {
+				fields := strings.Fields(line)
+				relfile, _ := filepath.Rel(workingDir, path)
+				pattern := fields[0]
+				if reldir := filepath.Dir(relfile); len(reldir) > 0 {
+					pattern = filepath.Join(reldir, pattern)
+				}
+				lockable := strings.Contains(line, LockableAttrib)
```

Comment:
Turns out there's a cheap way to do it, we're already splitting on whitespace so just need to check `fields[1:]` instead. `.gitattributes` always splits on plain whitespace (you have to use `[[:space:]]` in file patterns) so this works.